File: app/src/main/java/com/witype/Dragger/integration/HttpModel.java
package com.witype.Dragger.integration;

import com.witype.Dragger.entity.MobileDateUsageEntity;
import com.witype.Dragger.mvp.contract.ICacheRequestModel;
import com.witype.Dragger.mvp.contract.IRequestModel;
import com.witype.Dragger.mvp.model.ApiModel;
import com.witype.mvp.integration.CallDataModel;
import com.witype.mvp.integration.IRequestManager;
import com.witype.mvp.integration.RetrofitRequestManager;

import org.jetbrains.annotations.NotNull;

import javax.inject.Inject;

import io.reactivex.Observable;
import io.reactivex.schedulers.Schedulers;

public class HttpModel {

    @Inject
    RetrofitRequestManager retrofitRequestManager;

    @Inject
    public HttpModel() {
    }

    public Observable<MobileDateUsageEntity> getMobileDataUsage(String resourceId, int limit) {
        return retrofitRequestManager.createCache(ICacheRequestModel.class)
                .getMobileDataUsage(retrofitRequestManager.create(ApiModel.class)
                        .getMobileDataUsage(resourceId, limit)
                        .observeOn(Schedulers.io())
                        .subscribeOn(Schedulers.io())
                        .unsubscribeOn(Schedulers.io())
                );
    }

}

File: kotlindemo/src/main/java/com/witype/kotlindemo/integration/HttpModel.kt
package com.witype.kotlindemo.integration

import com.witype.kotlindemo.entity.MobileDateUsageEntity
import com.witype.kotlindemo.mvp.contract.ICacheRequestModel
import com.witype.kotlindemo.mvp.model.ApiModel
import com.witype.mvp.integration.RetrofitRequestManager
import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

open class HttpModel @Inject constructor() {

    @set:Inject
    lateinit var iRequestManager : RetrofitRequestManager

    fun getMobileDataUsage(resourceId: String, limit: Int): Observable<MobileDateUsageEntity> {
        return getRequestManager().createCache(ICacheRequestModel::class.java)
                .getMobileDataUsage(getRequestManager().create(ApiModel::class.java)
                        .getMobileDataUsage(resourceId, limit)
                        .observeOn(Schedulers.io())
                        .subscribeOn(Schedulers.io())
                        .unsubscribeOn(Schedulers.io())
                )
    }

    fun getRequestManager() : RetrofitRequestManager {
        return iRequestManager;
    }
}